internal/proto/data_coord.proto

```diff
@@ -397,6 +398,15 @@ message WatchChannelsResponse {
   common.Status status = 1;
 }
 
+message SetSegmentStateRequest {
+  int64 segment_id = 1;
+  common.SegmentState new_state = 2;
+}
+
+message SetSegmentStateResponse {
+  common.Status status = 1;
```

Comment:
if `common.Status` is the only result to return, the rpc signature could just return `common.Status`

Reply:
I meant to reserve this response type to support additional info. For example if we want to make this atomic we definitely need more info :)